newIDE/app/src/VariablesList/VariablesList.js

```diff
@@ -524,12 +527,28 @@ const VariablesList = (props: Props) => {
   );
 
   const [searchText, setSearchText] = React.useState<string>('');
-  const { onComputeAllVariableNames } = props;
+  const { onComputeAllVariableNames, onSelectedVariableChange } = props;
   const allVariablesNames = React.useMemo<?Array<string>>(
     () => (onComputeAllVariableNames ? onComputeAllVariableNames() : null),
     [onComputeAllVariableNames]
   );
-  const [selectedNodes, setSelectedNodes] = React.useState<Array<string>>([]);
+  // TODO Scroll to the initially selected variable and focus on the name.
+  const [selectedNodes, doSetSelectedNodes] = React.useState<Array<string>>(
+    props.initiallySelectedVariableName
+      ? [getNodeIdFromVariableName(props.initiallySelectedVariableName)]
+      : []
+  );
+  const setSelectedNodes = React.useCallback(
+    (nodes: Array<string> | ((nodes: Array<string>) => Array<string>)) => {
+      nodes = Array.isArray(nodes) ? nodes : nodes(selectedNodes);
+      doSetSelectedNodes(nodes);
+      if (onSelectedVariableChange) {
+        onSelectedVariableChange(nodes);
+      }
+    },
+    [onSelectedVariableChange, selectedNodes]
+  );
```

Comment:
Try:
```suggestion
  const setSelectedNodes = React.useCallback(
    (nodes: Array<string> | ((nodes: Array<string>) => Array<string>)) => {
      doSetSelectedNodes(selectedNodes => {
        const newNodes = Array.isArray(nodes) ? nodes : nodes(selectedNodes);
        if (onSelectedVariableChange) {
          onSelectedVariableChange(newNodes);
        }        
        return newNodes;
      });
    },
    [onSelectedVariableChange]
  );
```

to avoid relying on selectedNodes and having all the callbacks using setSelectedNodes to always change at each render whenever there is a change in selected nodes.